File: UEE-Pizza-hut-SL-main/backend/modals/promotion.js
const mongoose = require('mongoose');
const  Schema = mongoose.Schema;

const promotionSchema = new Schema({

    promotionTitle: {
        type: String,
        required: true
    },
    promoCode: {
        type: String,
        required: true
    },
    description:{
        type:String,
        required: true
    },
    discount:{
        type: Number, 
        required: true, 
    },
    imageUrl: {
        type: String,
        required: true
    },
})

const Promotion = mongoose.model("promotion", promotionSchema);
module.exports = Promotion;

File: UEE-Pizza-hut-SL-main/lib/model/Promotion.dart
import 'package:flutter/foundation.dart';

class Promotion {
  late final String promotionTitle;
  late final String promoCode;
  late final int discount;
  late final String imageUrl;
  late final List description;

  Promotion(
      {required this.promotionTitle,
      // required this.description,
      required this.discount,
      required this.imageUrl,
      required this.promoCode});

  factory Promotion.fromJson(Map<String, dynamic> json) {
    return Promotion(
        promotionTitle: json['promotionTitle'],
        // description: json['description'],
        discount: json['discount'],
        imageUrl: json['imageUrl'],
        promoCode: json['promoCode']);
  }
}
